src/SFA.DAS.EmployerAccounts.Jobs/RunOnceJobs/SeedAccountSignedAgreementsJob.cs

```diff
@@ -46,11 +46,12 @@ public async Task MigrateAgreements()
 
             _logger.LogInformation("Migrating signed agreements into the read store");
 
+            var readStoreAgreements = await CosmosDb.QueryableExtensions.ToListAsync(
+                _accountSignedAgreementsRepository.CreateQuery());
+
             foreach (var agreement in agreements)
```

Comment:
What do you think about this instead?

            // For All Agreements in the Employer DB
            // Find me all the agreements where
            // There is not any element in the financial agreement collection that matches AccountId Version and Agreement Type

            var differences = agreements.
                Where(agreement => !readStoreAgreements.Any
                    (fAgreement => fAgreement.AccountId == agreement.AccountId && fAgreement.AgreementVersion == agreement.Template.VersionNumber && fAgreement.AgreementType == agreement.Template.AgreementType.ToString()));

            // Depending on your preference for readability
            // The Inversion is
            // For All Agreements in the Employer DB
            // Find me all the agreements Where 
            // There is no financial agreement that matches accountID, Version and AgreementType in the financial agreement collection.

            var differences2 = agreements.
                Where(agreement => readStoreAgreements.
                All(fAgreement => fAgreement.AccountId != agreement.AccountId || fAgreement.AgreementVersion != agreement.Template.VersionNumber || fAgreement.AgreementType != agreement.Template.AgreementType.ToString()));


            foreach(var difference in differences)
            {
                var document = new AccountSignedAgreement(difference.AccountId, difference.Template.VersionNumber, difference.Template.AgreementType.ToString(), Guid.NewGuid());
                await _accountSignedAgreementsRepository.Add(document, null, CancellationToken.None);
            }